src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpOutlining.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.IntegrationTest.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Roslyn.VisualStudio.IntegrationTests.CSharp
+{
+    [Collection(nameof(SharedIntegrationHostFixture))]
+    public class CSharpOutlining : AbstractEditorTest
+    {
+        protected override string LanguageName => LanguageNames.CSharp;
+
+        public CSharpOutlining(VisualStudioInstanceFactory instanceFactory)
+            : base(instanceFactory, nameof(CSharpOutlining))
+        {
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Outlining)]
+        public void Outlining()
+        {
+            var input = @"
+using [|System;
+using System.Collections.Generic;
+using System.Text;|]
+
+namespace ConsoleApplication1[|
+{
+    public class Program[|
+    {
+        public static void Main(string[] args)[|
+        {
+            Console.WriteLine(""Hello World"");
+        }|]
+    }|]
+}|]";
+            MarkupTestFile.GetSpans(input, out var text, out ImmutableArray<TextSpan> spans);
+            VisualStudio.Editor.SetText(text);
+            VisualStudio.Workspace.WaitForAsyncOperations(FeatureAttribute.Outlining);
+            Assert.Equal(spans.OrderBy(s => s.Start), VisualStudio.Editor.GetOutliningSpans());
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Outlining)]
+        public void OutliningConfigChange()
+        {
+            var input = @"
+namespace ClassLibrary1[|
+{
+    public class Class1[|
+    {
+#if DEBUG
+{|Release:        void Foo(){|Debug:
+        {
+        }|}
+        
+        void Foo2(){|Debug:
+        {
+        }|}|}
+#else
+{|Debug:        void Bar(){|Release:
+        {
+        }|}|}
+#endif
+    }|]
+}|]";
+            MarkupTestFile.GetSpans(input, out var text, out IDictionary<string, ImmutableArray<TextSpan>> spans);
+            VisualStudio.Editor.SetText(text);
+
+            Verify(spans, "Release");
+            Verify(spans, "Debug");
+        }
+
+        private void Verify(IDictionary<string, ImmutableArray<TextSpan>> spans, string configuration)
```

Comment:
`VerifySpansInConfiguration` or something? Just `Verify` makes the test method hard to read. It's probably fine since there's only one usage, but if someone added more tests that use it it'd get annoying.